pubtools/_content_gateway/push_base.py

```diff
@@ -510,3 +555,74 @@ def process_file(self, file_item):
             raise CGWError("Cannot delete file: product version: '%s' product code: '%s' product: '%s', id is not set" %
                            (version_name, product_code, product_name))
         return file_id
+
+    def make_visible(self):
+        """
+        - Update the invisible attribute to false
+          once all the CGW operation completed successfully.
+
+        - This is needed to make the version and file records visible.
+        - The records get created with the flag invisible=True and
+          later on successful completion of CGW operations they get enabled.
+        
+        """
+
+        self.completed_operations.reverse()
+        for item in self.completed_operations:
+            if item.get('type') == 'product_version':
+                if item.get("operation") == "created":
+                    item["metadata"]["id"] = item["version_id"]
+                    item["metadata"]['invisible'] = False
```

Comment:
We talked about this one and this should be set to False if invisible is not set in definition in the yaml file, but otherwise it should respect user input

Reply:
Yes, this will be updated in my upcoming patch.